packages/drivers/iframe-driver/src/innerDocumentService.ts

```diff
@@ -22,16 +25,32 @@ export class InnerDocumentService implements IDocumentService {
     /**
      * Create a new InnerDocumentService
      */
-    public static async create(proxyObject: ICombinedDriver): Promise<InnerDocumentService> {
-        return new InnerDocumentService(proxyObject, proxyObject.clientId);
+    public static async create(
+        proxyObject: ICombinedDriver,
+        logger?: ITelemetryBaseLogger,
+    ): Promise<InnerDocumentService> {
+        return new InnerDocumentService(proxyObject, proxyObject.clientId, logger);
     }
 
+    private readonly logger: ITelemetryBaseLogger;
+
     private constructor(
         private readonly outerProxy: ICombinedDriver,
-        public clientId: string) { }
+        public clientId: string,
+        logger?: ITelemetryBaseLogger)
+    {
+        // outerProxy provides a logger as well, but we can create and use our own here
+        // for different functionality
+        this.logger = logger
+            ? ChildLogger.create(logger, "InnerIFrameDriver")
+            : new IFrameDebugLogger("InnerIFrameDriver");
+    }
 
     // TODO: Issue-2109 Implement detach container api or put appropriate comment.
     public get resolvedUrl(): IResolvedUrl {
+        const event = { category: "generic", eventName: "not implemented" };
+        this.outerProxy.logger.send(event);
+        this.logger.send(event);
```

Comment:
we should have to call two seperate loggers. there should just be a single combined logger